packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -865,59 +877,92 @@ export class GarbageCollector implements IGarbageCollector {
-        const explicitReferences: string[] = [];
-        for (const [, outboundRoutes] of Object.entries(this.gcDataFromLastRun.gcNodes)) {
-            explicitReferences.push(...outboundRoutes);
+        // Get Current and Previous graphs
+        const currentGraph = getOutboundGraph(Object.entries(currentGCData.gcNodes));
+        const previousGraph = previousGCData.gcNodes;
+
+        // Expected Routes = Routes in the Current Graph and not in the Previous Graph
+        const expectedRoutes: [string, string[]][] = [];
+        for(const [nodeId, currentReferences] of currentGraph) {
+            const prevOutboundReferences = getOutboundReferences(nodeId, previousGraph[nodeId]);
+            const expectedReferences: string[] =
+                currentReferences.filter((currentRef) => !prevOutboundReferences.includes(currentRef));
+            if(expectedReferences.length > 0) {
+                expectedRoutes.push([nodeId, expectedReferences]);
+            }
         }
-        this.referencesSinceLastRun.forEach((outboundRoutes: string[]) => {
-            explicitReferences.push(...outboundRoutes);
-        });
 
-        // Validate that the current reference graph doesn't have references that we are not already aware of. If this
-        // happens, it might indicate data corruption since we may delete objects prematurely.
-        currentReferences.forEach((route: string) => {
-            // Validate references for data stores only. Currently, layers below data stores don't have GC implemented
-            // so there is no guarantee their references will be notified.
-            if (isDataStoreNode(route) && !explicitReferences.includes(route)) {
-                /**
-                 * The following log will be enabled once this issue is resolved:
-                 * https://github.com/microsoft/FluidFramework/issues/8878.
-                 */
-                // We should ideally throw a data corruption error here. However, send an error for now until we have
-                // implemented sweep and have reasonable confidence in the sweep process.
-                // this.mc.logger.sendErrorEvent({
-                //     eventName: "gcUnknownOutboundRoute",
-                //     route,
-                // });
+        // Missing Routes = Expected Routes not in New Routes
+        const missingRoutes: [string, string[]][] = [];
+        for (const [nodeId, expectedReferences] of expectedRoutes) {
+            const newReferences = newRoutes.get(nodeId);
+            const missingReferences: string[] = [];
+            expectedReferences.forEach((expectedRef) => {
+                // Every time an expected route is found, there should be a log of the error.
+                // Eventually a data corruption assert should be thrown and the client should fail fast
+                if (newReferences === undefined || !newReferences.includes(expectedRef)) {
+                    this.mc.logger.sendPerformanceEvent({
```

Comment:
We cannot log this error yet because of the bug mentioned in the comment that was here before. You should only do it for testing behind a feature flag.